.clang-format

```diff
@@ -29,17 +28,20 @@ BreakStringLiterals: true
 ColumnLimit:     100
 CommentPragmas:  '^ IWYU pragma:'
 ConstructorInitializerAllOnOneLineOrOnePerLine: false
-ConstructorInitializerIndentWidth: 4
+ConstructorInitializerIndentWidth: 2
 ContinuationIndentWidth: 4
 Cpp11BracedListStyle: true
 DerivePointerAlignment: false
 DisableFormat:   false
 ExperimentalAutoDetectBinPacking: false
 ForEachMacros:   [ foreach, Q_FOREACH, BOOST_FOREACH ]
+IncludeBlocks: Regroup
```

Comment:
Do you have an example of how regrouped blocks will look like? Will it match the style guide?

Reply:
https://github.com/xbmc/xbmc/compare/master...Rechi:clang-format-includes
Moving PlatformDefs.h, system.h, system_gl.h and platform headers into seperate groups does not match our current coding style. I think it is useful to immediately see which headers are generic and which ones are platform specific.

Reply:
If you think it helps, why not. Style guide will have to be updated though.

Current style doc also distinguishes between C(++) standard library headers and other library headers, which we would have to remove. (fine by me though)